Snakefile

```diff
@@ -0,0 +1,92 @@
+import glob
+import json
+import os
+#Funtion to extract samples from input JSON file and generate necessary .txt files
+N_FILES_MAX_PER_SAMPLE = -1
+def extract_samples_from_json(json_file):
+    output_files = []
+    with open(json_file, "r") as fd:
+        data = json.load(fd)
+        for sample, conditions in data.items():
+            for condition, details in conditions.items():
+                #Creating a filename for the file and condition
+                sample_name = f"{sample}__{condition}"
+                output_files.append((sample, condition))
+                # Write paths to a .txt file with the correct path replacement
+                with open(f"sample_{sample_name}_paths.txt", "w") as path_file:
+                    paths = [file_info["path"] for file_info in details["files"]]
+                    path_file.write("\n".join(paths))
+                    paths = [
+                        file_info["path"].replace(
+                            "https://xrootd-local.unl.edu:1094//store/user/AGC/nanoAOD",
+                            "root://eospublic.cern.ch//eos/opendata/cms/upload/agc/1.0.0/"
+                        )
+                        for file_info in details["files"]
+                    ]
+    return output_files
+#Function to get file paths based on the index
+def get_file_paths(wildcards, max=N_FILES_MAX_PER_SAMPLE):
+    "Return list of at most MAX file paths for the given SAMPLE and CONDITION."
+    filepaths = []
+    with open(f"sample_{wildcards.sample}__{wildcards.condition}_paths.txt", "r") as fd:
+        filepaths = fd.read().splitlines()
+    # Using the index as the wildcard, creating a path for each file based on it's index
+    return [f"histograms/histograms_{wildcards.sample}__{wildcards.condition}__{index}.root" for index in range(len(filepaths))][:max]
+
+samples_conditions = extract_samples_from_json("nanoaod_inputs.json")
+
+rule all:
+    input:
+        "histograms_merged.root"
+
+rule process_sample_one_file_in_sample:
+    container:
+        "docker.io/reanahub/reana-demo-agc-cmc-ttbar-coffea:1.0.0"
```

Comment:
Note that the container is "-cms-", not "-cmc-".

Reply:
BTW I have built a new version of the container after your changes, and the workflow execution fails for me with the following error message:

```
papermill.exceptions.PapermillExecutionError:
---------------------------------------------------------------------------
Exception encountered at "In [2]":
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In [2], line 7
      5 import cloudpickle
      6 import correctionlib
----> 7 from coffea import processor
      8 from coffea.nanoevents import NanoAODSchema
      9 from coffea.analysis_tools import PackedSelection

File /opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/coffea/processor/__init__.py:6
      1 """A framework for analysis scale-out
      2
      3
      4 """
      5 from .processor import ProcessorABC
----> 6 from .dataframe import (
      7     LazyDataFrame,
      8 )
      9 from .helpers import Weights, PackedSelection
     10 from .executor import (
     11     IterativeExecutor,
     12     FuturesExecutor,
   (...)
     17     run_spark_job,
     18 )

File /opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/coffea/processor/dataframe.py:2
      1 from collections.abc import MutableMapping
----> 2 import uproot
      5 class LazyDataFrame(MutableMapping):
      6     """Simple delayed uproot reader (a la lazyarrays)
      7
      8     One can access branches either through ``df["bname"]`` or ``df.bname``, although
   (...)
     23             Additional metadata for the dataframe
     24     """

File /opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/uproot/__init__.py:78
     76 from uproot.version import __version__
     77 import uproot.const
---> 78 import uproot.extras
     79 import uproot.dynamic
     81 classes = {}

File /opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/uproot/extras.py:15
     12 import atexit
     13 import os
---> 15 import pkg_resources
     17 from uproot._util import parse_version
     20 def awkward():

File /opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/pkg_resources/__init__.py:95
     93 import packaging.utils
     94 import packaging.version
---> 95 from jaraco.text import drop_comment, join_continuation, yield_lines
     96 from platformdirs import user_cache_dir as _user_cache_dir
     98 if TYPE_CHECKING:

File /opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/setuptools/_vendor/jaraco/text/__init__.py:12
      9     from importlib_resources import files  # type: ignore
     11 from jaraco.functools import compose, method_cache
---> 12 from jaraco.context import ExceptionTrap
     15 def substitution(old, new):
     16     """
     17     Return a function that will perform a substitution on a string
     18     """

File /opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/setuptools/_vendor/jaraco/context.py:17
     13 from typing import Iterator
     16 if sys.version_info < (3, 12):
---> 17     from backports import tarfile
     18 else:
     19     import tarfile

ImportError: cannot import name 'tarfile' from 'backports' (/opt/micromamba/envs/analysis-systems/lib/python3.8/site-packages/backports/__init__.py)
```

Perhaps we should do some more version pinning in the various requirements files?